### PR TITLE
Supply genesis

### PR DESCRIPTION
**Pull request by alessio**
- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry: `clog add [section] [stanza] [message]`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ func NewGaiaApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest b
 	// initialized with tokens from genesis accounts.
 	app.mm.SetOrderInitGenesis(genaccounts.ModuleName, supply.ModuleName, distr.ModuleName,
```
Comment (colin-axner) [inline app/app.go]: shouldn't `supply.ModuleName` be removed after genaccounts - before distr?

Comment (alessio) [reply app/app.go]: of course it must be removed! thanks!

Comment (alessio) [reply app/app.go]: done btw

**Comment by jackzampolin** [comment]
This seems to be substantially similar to https://github.com/cosmos/gaia/pull/67

**Pull request closed without merging**